plugins/liquid/refined-types-plugin/src/main/kotlin/arrow/meta/plugins/liquid/phases/analysis/solver/ConstraintChecks.kt

```diff
@@ -631,7 +648,11 @@ private fun SolverState.checkBodyAgainstInvariants(
   return checkExpressionConstraints(newName, body, data).onEach {
     invariant?.let {
       val renamed = solver.renameObjectVariables(it, mapOf(RESULT_VAR_NAME to newName))
-      checkInvariant(renamed, data.context, element)
+      checkInvariant(
+        NamedConstraint("checkBodyInvariants: $RESULT_VAR_NAME renamed $newName", renamed),
```

Comment:
Should we introduce a name for `invariant` too?

Reply:
I introduced this on the last commit, but I realize invariants are never persisted in the annotations. I suppose they are only useful for checking the bodies of the local module.